lib/pbench/server/api/resources/query_apis/iteration_samples.py

```diff
@@ -89,31 +87,35 @@ def recurse(prefix, mappings, result):
                     recurse(f"{prefix}{p}.", m, result)
             elif mappings.get("type") == "keyword":
                 result.append(prefix[:-1])  # Remove the trailing dot, if any
-            elif mappings.get("type") == "text" and mappings.get("fields"):
-                for field in mappings.get("fields"):
-                    recurse(f"{prefix}{field}.", mappings.get("fields")[field], result)
+            else:
+                for f, v in mappings.get("fields", {}).items():
+                    recurse(f"{prefix}{f}.", v, result)
 
         recurse("", mappings, fields)
         return fields
 
     def assemble(self, json_data: JSON, context: CONTEXT) -> JSON:
         """
-        Construct a pbench search query for aggregating unique values for keyword
-        fields in the 'benchmark', 'sample', 'iterations' and other
+        Construct a pbench search query for aggregating a list of values for
+        keyword fields in the 'benchmark', 'sample', 'iterations' and 'run'
         sub-documents of result-data-sample index document that belong to the
```

Comment:
Rather than listing the sub-documents explicitly, you should probably just say "the white-listed sub-documents" (or "sub-documents in WHITELIST_AGGS_FIELDS"), since the definition of the list is elsewhere, and this text is unlike to be updated if the contents of the list are changed.